src/test/java/step4/CarTest.java

```diff
@@ -0,0 +1,89 @@
+package step4;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+
+class CarTest {
+
+    private static final int TEST_CAR_MOVE_STEP = 1;
+
+
+    // init
+    @DisplayName("init Class")
+    @ParameterizedTest
+    @CsvSource(value = {"carName:0", "anyTest:6", "carTest:9"}, delimiter = ':')
+    public void testClassInitialize(String carName, int location) {
+
+        Car car = new Car(carName, location, new CarMoveStrategy() {
+            @Override
+            public int getMoveCount() {
+                return TEST_CAR_MOVE_STEP;
+            }
+        });
```

Comment:
람다를 사용하면 간단하게 바꿀수있습니다 :)
```suggestion
        Car car = new Car(carName, location, () -> TEST_CAR_MOVE_STEP);;
```

Reply:
해당 위치의 역할에 맞는 Function IF 를 넣어야 한다고 생각했는데 막상 요래 해보니 필요한 간단한 기능은 바로 입력해버리는 방법도 좋네요 ^^